pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -970,7 +973,10 @@ private int numReplicasToLookFor() {
       }
       LOGGER.info("Committing segment {} at offset {} winner {}", _segmentName.getSegmentName(), offset, instanceId);
       _state = State.COMMITTING;
-      success = _segmentManager.commitSegment(_segmentName.getTableName(), _segmentName.getSegmentName(),
+      if (isSplitCommit) {
```

Comment:
Add a comment here that in case of splitCommit, the segment is uploaded to a unique file name indicated by segmentLocation, so we need to move the segment file to its permanent location first before committing the metadata.